server/driver/wivrn_session.cpp

```diff
@@ -162,13 +163,44 @@ xrt_result_t xrt::drivers::wivrn::wivrn_session::create_session(xrt::drivers::wi
 	if (self->hmd->face_tracking_supported)
 		usysds->base.base.static_roles.face = self->hmd.get();
 
-	if (self->left_hand)
+	xrt_device * active_left_hand = nullptr;
+	xrt_device * active_right_hand = nullptr;
+
+#ifdef WIVRN_FEAT_STEAMVR_LH
+	auto use_steamvr_lh = std::getenv("WIVRN_USE_STEAMVR_LH");
+	xrt_system_devices * lhdevs = NULL;
+
+	if (use_steamvr_lh && steamvr_lh_create_devices(&lhdevs) == XRT_SUCCESS)
+	{
+		U_LOG_W("Found %lu lighthouse devices:", lhdevs->xdev_count);
+		for (int i = 0; i < lhdevs->xdev_count; i++)
+		{
+			auto lhdev = lhdevs->xdevs[i];
+			devices->xdevs[n++] = lhdev;
+			switch (lhdev->device_type)
+			{
+				case XRT_DEVICE_TYPE_LEFT_HAND_CONTROLLER:
+					active_left_hand = lhdev;
+					break;
+				case XRT_DEVICE_TYPE_RIGHT_HAND_CONTROLLER:
+					active_right_hand = lhdev;
+					break;
+				default:
+					break;
+			}
+		}
+	}
+#endif
+
+	if (self->left_hand && !active_left_hand)
 	{
+		active_left_hand = self->left_hand.get();
 		devices->xdevs[n++] = self->left_hand.get();
 		usysds->base.base.static_roles.hand_tracking.left = self->left_hand.get();
 	}
-	if (self->right_hand)
+	if (self->right_hand && !active_right_hand)
 	{
+		active_right_hand = self->right_hand.get();
 		devices->xdevs[n++] = self->right_hand.get();
 		usysds->base.base.static_roles.hand_tracking.right = self->right_hand.get();
 	}
```

Comment:
we will eventually want to expose hand tracking, except nowadays OC completely breaks down when using WiVRn hand tracking + lighthouse controller.